### ISSUE TITLE
hi, i use apollo to train resnet18 at imagenet-1k datasets, using the parameters in README, but best acc is only 69.9% while the baseline acc is 70.5%. Is it the params setting problem or others?

### ISSUE THREAD

**Issue by JLtwoP** [XuezheMax/apollo#6]

**Comment by XuezheMax** [comment]
I have not evaluated Apollo on ResNet18.
Could you please provide the detailed setting, like how many epochs and what is the learning rate scheduler.
Moreover, what is your baseline method with 70.5% acc? 
According to the [PyTorch doc](https://pytorch.org/vision/stable/models.html), the acc of ResNet18 with SGD on ImageNet is around 69.76.

In addition, is 69.9% the acc of a single run or multiple runs?

**Comment by JLtwoP** [comment]
> I have not evaluated Apollo on ResNet18.
> Could you please provide the detailed setting, like how many epochs and what is the learning rate scheduler.
> Moreover, what is your baseline method with 70.5% acc?
> According to the [PyTorch doc](https://pytorch.org/vision/stable/models.html), the acc of ResNet18 with SGD on ImageNet is around 69.76.
> 
> In addition, is 69.9% the acc of a single run or multiple runs?

follow the params set as below, resnet18  val acc can be 70.5% by pytorch.
https://cv.gluon.ai/model_zoo/classification.html#

69.9% is the acc of multiple runs 192 bs/gpu on 8 gpus, 192*8，lr  follows your setting

**Comment by XuezheMax** [comment]
Hi, 
The link you pasted lists several resnet18 accs. Would you please paste your training script here? Thanks a lot.

In my paper, there are two lr decay settings: milestone and cosine. For your Apollo 69.9% runs, which one is used?

**Comment by JLtwoP** [comment]
> Hi,
> The link you pasted lists several resnet18 accs. Would you please paste your training script here? Thanks a lot.
> 
> In my paper, there are two lr decay settings: milestone and cosine. For your Apollo 69.9% runs, which one is used?

the r18 baseline training script is as bellow:
python train_imagenet.py \
  --rec-train /media/ramdisk/rec/train.rec --rec-train-idx /media/ramdisk/rec/train.idx \
  --rec-val /media/ramdisk/rec/val.rec --rec-val-idx /media/ramdisk/rec/val.idx \
  --model resnet18 \
  --lr 0.4 --lr-mode cosine --num-epochs 120 --batch-size 192--num-gpus 8 \
  --warmup-epochs 5 --weight_decay 0.0001 --momentum 0.9 \

69.9% acc is based on cosine lr setting

**Comment by XuezheMax** [comment]
Got it. Thanks!
I think I got the reason. Your batch size (192 * 8) is much larger than that in my setting (totally 256). When we are using a larger batch size, we probably need to increase the learning rate. For example, in your baseline, the lr of SGD is 0.4 rather than 0.1 in my setting. Please try to use larger lr for Apollo as well, e.g. 2.0 or 3.0.

**Comment by JLtwoP** [comment]
> Got it. Thanks!
> I think I got the reason. Your batch size (192 * 8) is much larger than that in my setting (totally 256). When we are using a larger batch size, we probably need to increase the learning rate. For example, in your baseline, the lr of SGD is 0.4 rather than 0.1 in my setting. Please try to use larger lr for Apollo as well, e.g. 2.0 or 3.0.

Thanks! I will try it~

**Comment by JLtwoP** [comment]
> > Got it. Thanks!
> > I think I got the reason. Your batch size (192 * 8) is much larger than that in my setting (totally 256). When we are using a larger batch size, we probably need to increase the learning rate. For example, in your baseline, the lr of SGD is 0.4 rather than 0.1 in my setting. Please try to use larger lr for Apollo as well, e.g. 2.0 or 3.0.
> 
> Thanks! I will try it~

**Comment by JLtwoP** [comment]
hi, I try the experiment as your suggest. The result is that apollo can improve acc by 0.2 on resnet18.

Another question is whether you have tried BFGS/k-FAC. If you have tried it, how is its generalization？

**Comment by XuezheMax** [comment]
Thanks for letting me know your results!
For BFGS/K-FAC, I never tried them on large neural networks, due to the costs.
In the appendix of my paper, I tried SldBFGS on a toy CNN.

**Comment by XuezheMax** [comment]
@JLtwoP 

For your reference, I trained ResNet-18 with Apollo with the same setting reported in the paper:
epochs=120, batch=256 (128/gpu * 2), lr=1.0, weight-decay=1e-4.
With the two learning rate decay scheduler (milestone & cosine), Apollo obtains:

| Method     |  milestone      |  cosine      |
| :--------- | :------------: | :---------: |
| SGD          |  70.12             |  70.66        |
| Apollo       |  70.56            |   71.10        |
